build-tools/src/main/java/software/amazon/awssdk/buildtools/findbugs/DisallowMethodCall.java

```diff
@@ -44,6 +45,18 @@ public class DisallowMethodCall extends OpcodeStackDetector {
         PROHIBITED_METHODS.add(new SimpleEntry<>("software/amazon/awssdk/http/SdkHttpRequest", "rawQueryParameters"));
         PROHIBITED_METHODS.add(new SimpleEntry<>("software/amazon/awssdk/http/SdkHttpFullRequest", "rawQueryParameters"));
         PROHIBITED_METHODS.add(new SimpleEntry<>("software/amazon/awssdk/http/SdkHttpFullRequest$Builder", "rawQueryParameters"));
+
+        // Blocking calls in async code path
+        PROHIBITED_ASYNC_BLOCKING_METHODS.add(new SimpleEntry<>("java/lang/Thread", "sleep"));
+        PROHIBITED_ASYNC_BLOCKING_METHODS.add(new SimpleEntry<>("java/util/concurrent/CompletableFuture", "join"));
+        PROHIBITED_ASYNC_BLOCKING_METHODS.add(new SimpleEntry<>("java/util/concurrent/CompletableFuture", "get"));
+        PROHIBITED_ASYNC_BLOCKING_METHODS.add(new SimpleEntry<>(
+            "software/amazon/awssdk/utils/CompletableFutureUtils", "joinLikeSync"));
+        PROHIBITED_ASYNC_BLOCKING_METHODS.add(new SimpleEntry<>(
```

Comment:
do we also want to include wait() - https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#:~:text=wait,-public%20final%20void&text=()%20throws%20InterruptedException-,Causes%20the%20current%20thread%20to%20wait%20until%20another%20thread%20invokes,must%20own%20this%20object's%20monitor.

Reply:
added Object.wait(), good call